### ISSUE TITLE
[UX] Indicate (and lock?) overridden settings in forms.

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#4331]
In #2805 we have made it possible to override any config setting in settings.php; this issue here is a follow-up, to indicate overridden settings in the UI, and to have them "locked".

Screenshot from @herbdool's PoC PR:

![Screenshot from 2019-04-23 16-23-48](https://user-images.githubusercontent.com/1071483/56613359-7caa5f80-65e4-11e9-8fb0-9490e0f80032.png)

---
PoC PR by @herbdool: https://github.com/backdrop/backdrop/pull/2607

**Comment by herbdool** [comment]
Sadly I don't think my approach here is ideal because it would require adding two lines to every form element, including all contribute modules. Since we've merged an approach that allows any element to be overridden we need a more general approach here too. Need a way to match a form element with a config variable. I'm not sure how.

**Comment by klonos** [comment]
You are right; for posterity, there is conversation re matching form elements to config keys (so we can check whether they've been overridden or not) after comment https://github.com/backdrop/backdrop-issues/issues/2805#issuecomment-485322643
